apis/src/components/organisms/analysis/history.rs

```diff
@@ -0,0 +1,195 @@
+}
+
+#[component]
+pub fn History(#[prop(optional)] extend_tw_classes: &'static str) -> impl IntoView {
+    let analysis = expect_context::<AnalysisSignal>().0;
+    let prev_button = create_node_ref::<html::Button>();
+    let next_button = create_node_ref::<html::Button>();
+    let window = use_window();
+    let active = Signal::derive(move || {
+        if window.is_some() {
+            window
+                .as_ref()
+                .expect("window to exist")
+                .document()
+                .expect("window to have a document")
+                .query_selector(".bg-orange-twilight")
+                .expect("to have an Element")
+        } else {
+            None
+        }
+    });
+    _ = use_event_listener(document().body(), keydown, move |evt| {
+        if evt.key() == "ArrowLeft" {
+            evt.prevent_default();
+            if let Some(prev) = prev_button.get_untracked() {
+                prev.click()
+            };
+        } else if evt.key() == "ArrowRight" {
+            evt.prevent_default();
+            if let Some(next) = next_button.get_untracked() {
+                next.click()
+            };
+        }
+    });
+    let focus = Callback::new(move |()| {
+        if let Some(elem) = active.get_untracked() {
+            elem.scroll_into_view_with_bool(false);
+        }
+    });
+    let walk_tree = move || {
+        let tree = analysis.get().unwrap().tree;
+        if tree.get_nodes().is_empty() {
+            return None;
+        }
+        let node_order = tree.traverse(TraversalStrategy::PostOrder, &0);
+        let mut content = None;
+        let mut branches = HashMap::<usize, View>::new();
+        let mut visited = vec![false; tree.get_nodes().len()];
+        for node_id in node_order {
+            let mut current_node = node_id;
+            loop {
+                if visited[current_node] {
+                    break;
+                }
+                visited[current_node] = true;
+                let node = tree.get_node(&current_node).unwrap();
+                let children = node.get_children();
+                let parent = node.get_parent();
+                let mut first_child = usize::MAX;
+                let parent_deg = if let Some(parent) = parent {
+                    first_child = tree.get_node(&parent).unwrap().get_children()[0];
+                    tree.get_node_degree(&parent)
+                } else {
+                    0
+                };
+                content = Some(if children.len() > 1 {
+                    // Gather all children but the first
+                    let inner = store_value(
+                        children
+                            .iter()
+                            .skip(1)
+                            .map(|child| branches.get(child))
+                            .collect::<Vec<_>>()
+                            .into_view(),
+                    );
+                    //unwrap is safe because we know there is a first child
```

Comment:
all children branches will surly be processed due to Post Order tree traversal, in fact in the above map function one may also safely place an unwrap and see the code never panic. I was putting this as a kind of "assert" to verify I will draw the child.

All that said, this unwrap is superfluous since one may pass a None to render so I will just remove :D